pkg/ext/stores/tokens/tokens.go

```diff
@@ -0,0 +1,1459 @@
+	secret.StringData[FieldLabels] = string(labelBytes)
+	secret.StringData[FieldAnnotations] = string(annotationBytes)
+
+	// spec
+	enabled := token.Spec.Enabled == nil || *token.Spec.Enabled
+
+	secret.StringData[FieldUserID] = token.Spec.UserID
+	secret.StringData[FieldTTL] = fmt.Sprintf("%d", ttl)
+	secret.StringData[FieldEnabled] = fmt.Sprintf("%t", enabled)
+	secret.StringData[FieldDescription] = token.Spec.Description
+	secret.StringData[FieldKind] = token.Spec.Kind
+	secret.StringData[FieldPrincipalID] = token.Spec.PrincipalID
+
+	lastUsedAsString := ""
+	if token.Status.LastUsedAt != nil {
+		lastUsedAsString = token.Status.LastUsedAt.Format(time.RFC3339)
+	}
+
+	// status
+	secret.StringData[FieldHash] = token.Status.TokenHash
+	secret.StringData[FieldLastUpdateTime] = token.Status.LastUpdateTime
+	secret.StringData[FieldLastUsedAt] = lastUsedAsString
+
+	// FUTURE ((USER ACTIVITY)) change as required by the field type
+	// secret.StringData[FieldIdleTimeout] = fmt.Sprintf("%d", token.Status.IdleTimeout)
+
+	// Note:
+	// - While the derived expiration data is not stored, the user-related information is.
+	// - The expiration data is computed trivially from spec and resource data.
+	// - Getting the user-related information on the other hand is expensive.
+	// - It is better to cache it in the backing secret
+
+	secret.StringData[FieldAuthProvider] = token.Status.AuthProvider
+	secret.StringData[FieldDisplayName] = token.Status.DisplayName
+	secret.StringData[FieldUserName] = token.Status.UserName
+
+	return secret, nil
+}
+
+// tokenFromSecret converts the secret argument (retrieved from the k8s store)
+// into the equivalent token.
+func tokenFromSecret(secret *corev1.Secret) (*ext.Token, error) {
+	// Basic result. This will be incrementally filled as data is extracted from the secret.
+	// On error a partially filled token is returned.
+	// See the token store `Delete` (marker **) for where this is important.
+	token := &ext.Token{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Token",
+			APIVersion: "ext.cattle.io/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              secret.Name,
+			CreationTimestamp: secret.CreationTimestamp,
+		},
+	}
+
+	token.Spec.Description = string(secret.Data[FieldDescription])
+	token.Spec.Kind = string(secret.Data[FieldKind])
+
+	token.Status.DisplayName = string(secret.Data[FieldDisplayName])
+	token.Status.UserName = string(secret.Data[FieldUserName])
+
+	userId := string(secret.Data[FieldUserID])
+	if userId == "" {
+		return token, fmt.Errorf("user id missing")
+	}
+	token.Spec.UserID = userId
+
+	// spec
+	enabled, err := strconv.ParseBool(string(secret.Data[FieldEnabled]))
+	if err != nil {
+		return token, err
+	}
+	token.Spec.Enabled = &enabled
+
+	ttl, err := strconv.ParseInt(string(secret.Data[FieldTTL]), 10, 64)
+	if err != nil {
+		return token, err
+	}
+	// inject default on retrieval
+	if ttl == 0 {
+		ttl = ThirtyDays
+	}
+	token.Spec.TTL = ttl
+
```

Comment:
Placeholder for integration of PR #48778 here.

BEWARE. This code may have to handle rancher instances containing ext tokens without idle information, without crashing. I.e. when the work on user activity merges later than this PR. If that is true insertion a default when the information is not present, instead of returning an error, may be the more proper way of doing things.
```
idle, err := strconv.ParseInt(string(secret.Data[FieldIdleTimeout]), 10, 64)
if err != nil {
	return token, err
}
token.Status.IdleTimeout = idle
```